src/libraries/Microsoft.PowerFx.Connectors/FxParameterLocation.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    internal enum FxParameterLocation
+    {
+        Query,
+        Header,
+        Path,
+        Cookie,
+        Body
+    }
+
+    internal static partial class FxExtentions
+    {
+        public static FxParameterLocation? ToFxParameterLocation(this ParameterLocation? location)
+        {
+            return location switch
+            {
+                null => null,
+                ParameterLocation.Header => FxParameterLocation.Header,
+                ParameterLocation.Query => FxParameterLocation.Query,
+                ParameterLocation.Path => FxParameterLocation.Path,
+                ParameterLocation.Cookie => FxParameterLocation.Cookie,
+                _ => throw new NotImplementedException("Unknown location")
```

Comment:
>"Unknown location" [](http://example.com/codeflow?start=55&length=18)

in general, would be good ot include location in this message for diagnostics. 

 throw new NotImplementedException($"Unknown location {location}")

{} will handle null.